integration-tests/bats/schema-changes.bats

```diff
@@ -254,3 +254,452 @@ SQL
+    dolt sql -q "create table t(pk int PRIMARY KEY, val1 int, val2 int);"
+    dolt sql -q "alter table t add index (val2)"
+
+    dolt sql -q "insert into t values (1,1,1), (2,2,2)"
+    run dolt sql -q "ALTER TABLE t DROP PRIMARY KEY"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM t ORDER BY pk" -r csv
+    [[ "$output" =~ "1,1,1" ]] || false
+    [[ "$output" =~ "2,2,2" ]] || false
+
+    run dolt sql -q "INSERT INTO t VALUES (3, 3, 3)"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM t ORDER BY pk" -r csv
+    [[ "$output" =~ "1,1,1" ]] || false
+    [[ "$output" =~ "2,2,2" ]] || false
+    [[ "$output" =~ "3,3,3" ]] || false
+
+    run dolt sql -q "describe t"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "MUL" ]] || false
+}
+
+@test "schema-changes: drop on table with no primary key correctly errors" {
+     dolt sql -q "create table t(pk int, val1 int, val2 int);"
+
+     run dolt sql -q "ALTER TABLE t DROP PRIMARY KEY"
+     [ "$status" -eq 1 ]
+     [[ "$output" =~ "error: can't drop 'PRIMARY'; check that column/key exists" ]] || false
+}
+
+@test "schema-changes: drop primary key with auto increment throws an error" {
+    dolt sql -q "create table t(pk int PRIMARY KEY AUTO_INCREMENT, val1 int, val2 int)"
+    run dolt sql -q "alter table t drop primary key"
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "error: incorrect table definition: there can be only one auto column and it must be defined as a key" ]] || false
+}
+
+@test "schema-changes: ff merge with primary key schema differences correctly works" {
+    dolt sql -q "create table t(pk int PRIMARY KEY, val1 int, val2 int)"
+    dolt sql -q "INSERT INTO t values (1,1,1)"
+
+    dolt commit -am "cm1"
+    dolt checkout -b test
+    dolt sql -q "ALTER TABLE t drop PRIMARY key"
+    dolt add .
+    dolt commit -am "cm2"
+    dolt checkout master
+
+    run dolt merge test
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "describe t;"
+    ! [[ "$output" =~ 'PRI' ]] || false
+}
+
+@test "schema-changes: merge on primary key schema differences throws an error" {
+    dolt sql -q "create table t(pk int PRIMARY KEY, val1 int, val2 int)"
+    dolt sql -q "INSERT INTO t values (1,1,1)"
+    dolt commit -am "cm1"
+    dolt checkout -b test
+
+    dolt sql -q "ALTER TABLE t drop PRIMARY key"
+    dolt add .
+
+    run dolt status
+    [[ "$output" =~ 'Changes to be committed' ]] || false
+    [[ "$output" =~ ([[:space:]]*modified:[[:space:]]*t) ]] || false
+    ! [[ "$output" =~ 'deleted' ]] || false
+    ! [[ "$output" =~ 'new table' ]] || false
+
+    dolt commit -m "cm2"
+    dolt checkout master
+
+    dolt sql -q "INSERT INTO t values (2,2,2)"
+    dolt commit -am "cm3"
+
+    run dolt merge test
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ 'different primary key definitions for our column pk and their column pk' ]] || false
+}
+
+@test "schema-changes: same as the previous test but in the opposite direction" {
```

Comment:
needs a more descriptive title, tests can be inserted between these two

Reply:
Fair enough haha